### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by pyr0gan**
There is an issue I can't resolve with the Secret test suite, I started to get an error: `Error: Unsupported Pubkey type. Amino prefix: eb5ae98721` where it had previously been working. I am also not sure how to mock a faucet, as there is no programmatic way to airdrop tokens on Secret.

The Polygon test case works ✅

Please have a look @zurgl @guillaumegaluz

### CODE REVIEW

**Comment by zurgl** [comment]
@pyr0gan I've fixed the **module path aliases issue**
The mapping is under the jest.config.js is you want to know more.
btw, I canno't run the secret test.

```text
[zu@zu new-learn]$ npx jest ./__test__/secret.test.ts 
 FAIL  __test__/secret.test.ts
  ● Test suite failed to run

    Cannot find module '__mocks__/secretfaucet' from '__test__/secret.test.ts'

      12 | import dotenv from 'dotenv';
      13 |
    > 14 | import {SCRT_AMOUNT} from '__mocks__/secretfaucet';
         | ^
      15 |
      16 | dotenv.config({path: '.env.local'});
      17 |

      at Resolver.resolveModule (node_modules/jest-resolve/build/resolver.js:322:11)
      at Object.<anonymous> (__test__/secret.test.ts:14:1)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        0.683 s
Ran all test suites matching /.\/__test__\/secret.test.ts/i.
```

Any ideas ?

File: __test__/polygon.test.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  const url = getNodeURL();
+  const provider = new ethers.providers.JsonRpcProvider(url, 'any');
+  const blockHeight = await provider.getBlockNumber();
+  const blockInfo = await provider.getBlockWithTransactions(blockHeight);
+  return {blockInfo};
+}
+
+describe('Polygon backend tests', () => {
+  // Avoid jest open handle error
+  jest.setTimeout(30000);
```
Comment (guillaumegaluz) [inline __test__/polygon.test.ts:27]: 30 seconds? Why? If we do this other test cases, isn't this gonna block the test run for minutes and minutes?

File: __test__/polygon.test.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+dotenv.config({path: '.env.local'});
+
+// Avoid jest open handle error
+afterAll(async () => {
+  await new Promise<void>((resolve) => setTimeout(() => resolve(), 10000));
+});
+
+async function query_step() {
+  const url = getNodeURL();
```
Comment (guillaumegaluz) [inline __test__/polygon.test.ts:18]: Shouldn't we be passing arguments to this function? Also, does this mean we will be hitting devnet when running tests?

**Review by guillaumegaluz — COMMENTED** [review]
The broader question I have is: what kind of behaviors from the app to do want to ensure we don't break as we make future changes?

Right now it feels like you're testing the behavior of the ethers library, not of our app.

**Comment by zurgl** [comment]
@guillaumegaluz @pyr0gan Sorry guys, I was working on the main branch but without taking care this my work have merged commir **af890f0** then when I've pushed to main it as merged the PR too as the commit **af890f0** was ahead of this PR.

**Comment by pyr0gan** [comment]
> @guillaumegaluz @pyr0gan Sorry guys, I was working on the main branch but without taking care that my work included commit **[af890f0](https://github.com/figment-networks/learn-web3-dapp/commit/af890f004dfb07a57a75e6bd46887802a398f2e8)** then when I've pushed to main it as merged the PR too as the commit **[af890f0](https://github.com/figment-networks/learn-web3-dapp/commit/af890f004dfb07a57a75e6bd46887802a398f2e8)** was ahead of this PR.

😱 Probably best to just delete my tests, they're incomplete anyway.

**Pull request merged**